evidence/pool_test.go

```diff
@@ -53,18 +55,22 @@ func initializeValidatorState(valAddr []byte, height int64) dbm.DB {
 
 func TestEvidencePool(t *testing.T) {
 
-	valAddr := []byte("val1")
-	height := int64(5)
-	stateDB := initializeValidatorState(valAddr, height)
-	evidenceDB := dbm.NewMemDB()
-	pool := NewPool(stateDB, evidenceDB)
+	var (
+		valAddr      = []byte("val1")
+		height       = int64(5)
+		stateDB      = initializeValidatorState(valAddr, height)
+		evidenceDB   = dbm.NewMemDB()
+		pool         = NewPool(stateDB, evidenceDB)
+		evidenceTime = time.Date(2019, 1, 1, 0, 0, 0, 0, time.UTC)
+	)
 
-	goodEvidence := types.NewMockGoodEvidence(height, 0, valAddr)
-	badEvidence := types.MockBadEvidence{MockGoodEvidence: goodEvidence}
+	goodEvidence := types.NewMockEvidence(height, time.Now(), 0, valAddr)
+	badEvidence := types.NewMockEvidence(height, evidenceTime, 0, valAddr)
```

Comment:
@melekes I think this case tests trying to add old evidence, do you want another test case?

Reply:
sure

Reply:
added a test for adding evidence